### PR TITLE
Sentry improvements

### PR DESCRIPTION
**Pull request by lkostrowski**
## Context

Sentry is a first layer of our monitoring/observability (second is Datadog). We need to explore its features and establish process:
1. Alerts of errors - when to react, triage
2. Best possible debugging

First step is to unify Sentry across the packages and improve the configuration

Next steps will be connecting special events like messages, warnings and breadcrumbs (unhandled errors are automatic)

## Scope of the PR

1. Update Sentry version - unify in all packages
4. Updated Sentry config and unified in all packages
5. Set Sentry release to match app version instead of the commit

## Checklist
* [x] `.github/dependabot.yaml` is up-to date.
* [x] I added changesets and [read good practices](/.changeset/README.md).

TODO

- [x] Update urql to v4 to fix build problems https://github.com/saleor/apps/issues/655

### CODE REVIEW

**Review by krzysztofwolski — CHANGES_REQUESTED** [review]

File: apps/cms/src/pages/home.tsx [hunk 1]
```diff
@@ -25,12 +27,38 @@ const useStyles = makeStyles({
...
+                  level: "warning",
+                  data: {},
+                  message: "User clicked error test button",
+                });
+
+                Sentry.captureException(new Error("Test event from CMS app"));
+              });
+            }}
+          >
+            TEST ERROR SENTRY
```
[the remark this answers is not in the document]
Comment (lkostrowski) [reply apps/cms/src/pages/home.tsx:60]: of course not 😆 fixing

[the remark this answers is not in the document]
Comment (lkostrowski) [reply apps/cms/src/pages/home.tsx:60]: however @krzysztofwolski & @peelar please take a look at this example how to create a meaningful Sentry errors

```
Sentry.withScope((scope) => {
                scope.setUser({
                  segment: appBridgeState?.domain,
                });

                scope.setContext("app-bridge-state", {
                  appId: appBridgeState?.id,
                  locale: appBridgeState?.locale,
                });

                scope.addBreadcrumb({
                  level: "warning",
                  data: {},
                  message: "User clicked error test button",
                });

                Sentry.captureException(new Error("Test event from CMS app"));
```

[the remark this answers is not in the document]
Comment (lkostrowski) [reply apps/cms/src/pages/home.tsx:60]: We can work on some abstractions later (useSentry hook with its context etc)

[the remark this answers is not in the document]
Comment (lkostrowski) [reply apps/cms/src/pages/home.tsx]: fixed

File: apps/cms/src/providers/GraphQLProvider.tsx [hunk 2]
```diff
@@ -8,10 +8,20 @@ export function GraphQLProvider(props: PropsWithChildren<{}>) {
...
   });
+=======
+  if (!saleorApiUrl) {
+    return <>{props.children}</>;
+  }
+
+  const client = createClient(saleorApiUrl, async () => ({
+    token: appBridgeState?.token!,
+  }));
+>>>>>>> ab0d55dd (Update Sentry in CMS)
```
Comment (peelar) [inline apps/cms/src/providers/GraphQLProvider.tsx]: something is no yes here

**Review by peelar — COMMENTED** [review]
We could use some context about what exactly are we improving in our Sentry setup

**Review by peelar — APPROVED** [review]

**Pull request merged**